calculate_trigger.py

```diff
@@ -228,18 +228,19 @@ def calculate_trigger_addresses(fault_list, goldenrun_tb_exec, goldenrun_tb_info
                 continue
 
             found = False
-            for tdict in cachelist:
-                if (
-                    tdict["faultaddress"] == fault.address
-                    and tdict["faultlifespan"] == fault.lifespan
-                    and tdict["triggerhitcounter"] == fault.trigger.hitcounter
-                    and tdict["triggeraddress"] == fault.trigger.address
-                ):
-                    fault.trigger.address = tdict["answer"][0]
-                    fault.trigger.hitcounter = tdict["answer"][1]
-                    fault.lifespan = tdict["answer"][2]
-                    found = True
-                    break
+            tdict = cache_dict.get(fault.address)
+            if (
+                tdict is not None
+                and tdict["faultlifespan"] == fault.lifespan
+                and tdict["triggerhitcounter"] == fault.trigger.hitcounter
+                and tdict["triggeraddress"] == fault.trigger.address
+            ):
+                fault.trigger.address = tdict["answer"][0]
+                fault.trigger.hitcounter = tdict["answer"][1]
+                fault.lifespan = tdict["answer"][2]
+                found = True
+                break
```

Comment:
If I am not mistaken the break statement was to break the for-loop for the cachelist.
Now the break statement breaks an "upper" for-loop.